index_impl.go

```diff
@@ -530,10 +530,23 @@ func (i *indexImpl) SearchInContext(ctx context.Context, req *SearchRequest) (sr
 			} else if facetRequest.DateTimeRanges != nil {
 				// build date range facet
 				facetBuilder := facet.NewDateTimeFacetBuilder(facetRequest.Field, facetRequest.Size)
-				dateTimeParser := i.m.DateTimeParserNamed("")
+				dateTimeParserName := defaultDateTimeParser
 				for _, dr := range facetRequest.DateTimeRanges {
-					start, end := dr.ParseDates(dateTimeParser)
-					facetBuilder.AddRange(dr.Name, start, end)
+					if dr.DateTimeParser != "" {
+						dateTimeParserName = dr.DateTimeParser
```

Comment:
Hmm could there be a situation here, where one of the facet range has a datetime parser defined while the other doesn't. In this situation, we should fall back to using the defaultDateTimeParser for the 2nd facet as opposed to previous facet's parser.

This code will need correction to ..

```
dateTimeParsersName := defaultDateTimeParser
if dr.DateTimeParser != "" {
    dateTimeParserName = dr.DateTimeParser
}
```

Reply:
ok yea my bad